### PR TITLE
Skip stats under error conditions.

### PR DESCRIPTION
**Pull request by thakurajayL**

### CODE REVIEW

File: stats/stats.go [hunk 1]
```diff
@@ -178,36 +178,42 @@ func readStats() {
 		case AUTH_REQ_IN:
 			logger.StatsLog.Infoln("Received Event: AUTH_REQ_IN: ", m)
 			t := popTrans(m.Id) // remove MSG in trans but use the time msg was received
-			m.T = t.T
-			ue := getUe(m.Supi)
-			ue.CReg.AuthReqInTime = m.T
-			x := m.T.Sub(ue.CReg.RegReqOutTime)
-			ue.CReg.RegReqAuthReq = x.Microseconds()
-			logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
+			if t != nil {
```
Comment (gab-arrobo) [inline stats/stats.go:181]: Do you know why the transaction (Id) does not already exist in the `StatisticsEvent`?

Comment (gab-arrobo) [reply stats/stats.go:181]: After further debugging, I see that the same transaction (Id=14) gets "popped" twice in the stats, which lead to `nil` pointer. Transaction 14 comes from `HandleInitialContextSetupRequest`

```
Received Event: PDU_SESS_ACC_IN, Id: 9
Received Event: UE_CTX_CMD_IN, Id: 12
Received Event: ICS_REQ_IN, Id: 14
Received Event: SVC_ACCEPT_IN, Id: 14
```

File: stats/stats.go [hunk 2]
```diff
@@ -178,36 +178,42 @@ func readStats() {
...
+				m.T = t.T
+				ue := getUe(m.Supi)
+				ue.CReg.AuthReqInTime = m.T
+				x := m.T.Sub(ue.CReg.RegReqOutTime)
+				ue.CReg.RegReqAuthReq = x.Microseconds()
+				logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
+			}
```
Comment (gab-arrobo) [inline stats/stats.go:188]: What about just adding the following right after `t := popTrans(m.Id)` (similar to the approach we suggested in the AMF PR)
```go
			if t == nil {
				continue
			}
```
```suggestion
			if t == nil {
				continue
			}
			m.T = t.T
			ue := getUe(m.Supi)
			ue.CReg.AuthReqInTime = m.T
			x := m.T.Sub(ue.CReg.RegReqOutTime)
			ue.CReg.RegReqAuthReq = x.Microseconds()
			logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
```

Comment (gab-arrobo) [reply stats/stats.go:188]: Any comment/update about this suggestion?

Comment (gab-arrobo) [reply stats/stats.go:188]: @thakurajayL, any comment/update about this suggestion?

File: stats/stats.go (hunk 1 above)
Comment (thakurajayL) [reply stats/stats.go]: Yes current design needs some changes to handle few corner cases and hence it crashes if transaction not found. I would say at least  gnbsim should not crash if no transaction found.

**Review by gab-arrobo — APPROVED** [review]

**Pull request merged**